### PR TITLE
Add role to install lua job submit plugin

### PR DESCRIPTION
**Pull request by eesaanatluri**
Closes https://github.com/jprorama/CRI_XCBC/issues/479

### CODE REVIEW

File: roles/job_submit_plugin/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    backup: yes
+
+- name: Download job_submit.lua from GitHub Slurm repository
+  ansible.builtin.get_url:
+    url: "{{ job_submit_lua_url }}"
+    dest: "{{ job_submit_lua_dest }}"
+    mode: '0644'
+
+- name: reconfigure compute nodes with new config
+  ansible.builtin.command: scontrol reconfigure
```
Comment (eesaanatluri) [inline roles/job_submit_plugin/tasks/main.yml:18]: Do we really need to reconfigure, if we run the role before the `cod_compute_node` role is run?

File: group_vars/all [hunk 2]
```diff
@@ -351,3 +351,8 @@
   gpfs_groups:
     - {"name": "gpfs4", "authorized_keys":"/gpfs4/data/user/home/$DOWNSTREAM_USER/.ssh/authorized_keys", "host": "login001", "private_key":"/gpfs4/data/user/home/$DOWNSTREAM_USER/.ssh/id_ecdsa"}
     - {"name": "gpfs5", "authorized_keys":"/gpfs5/data/user/home/$DOWNSTREAM_USER/.ssh/authorized_keys", "host": "login002", "private_key":"/gpfs5/data/user/home/$DOWNSTREAM_USER/.ssh/id_ecdsa"}
+
+# job_submit_plugin
+  enable_job_submit_plugin: true
+  job_submit_lua_url: "https://gitlab.rc.uab.edu/rc/slurm-factory/-/blob/main/etc/slurm/job_submit.lua"
```
Comment (diedpigs) [inline group_vars/all:357]: Use this link instead:
https://gitlab.rc.uab.edu/rc/slurm-factory/-/raw/main/etc/slurm/job_submit.lua

File: group_vars/all [hunk 3]
```diff
@@ -351,3 +351,8 @@
...
+  job_submit_lua_dest: "/etc/slurm/slurm.conf"
```
Comment (diedpigs) [inline group_vars/all:358]: Wrong file name

File: group_vars/all (hunk 2 above)
Comment (eesaanatluri) [reply group_vars/all]: fixed ca467f585b472f86d47b568ff2764ebbbe1e616b

File: group_vars/all (hunk 3 above)
Comment (eesaanatluri) [reply group_vars/all]: fixed in ca467f585b472f86d47b568ff2764ebbbe1e616b

**Pull request merged**